### PR TITLE
[Intel MKL] Replace redundant attribute function with a generic function.

### PR DESCRIPTION
**Pull request by Zantares**
In MKL layout pass, use an iterator to get all attributes from original node, and copy them to the node builder instead of specific function. It helps to remove redundant code.
TF will throw an error if the two nodes have different attributes, so it can be easily found when copied wrong attributes.

modified:
- tensorflow/core/graph/mkl_layout_pass.cc

Signed-off-by: Lu Teng teng.lu@intel.com

### CODE REVIEW

File: tensorflow/core/graph/mkl_layout_pass.cc [hunk 1]
```diff
@@ -2355,6 +2308,21 @@ void MklLayoutRewritePass::AddWorkSpaceEdgeIfNeeded(
...
+                                        bool change_format) {
+  string name;
+  AttrSlice attr_list(orig_node->def());
+
+  auto iter = attr_list.begin();
+  while (iter != attr_list.end()) {
+    name = iter->first;
+    auto attr = iter->second;
+    nb->Attr(name, attr);
+    iter++;
```
Comment (penpornk) [inline tensorflow/core/graph/mkl_layout_pass.cc]: Nit:
```suggestion
    ++iter;
```

**Review by penpornk — CHANGES_REQUESTED** [review]
This is awesome! Thank you! :)

File: tensorflow/core/graph/mkl_layout_pass.cc (hunk 1 above)
Comment (Zantares) [reply tensorflow/core/graph/mkl_layout_pass.cc]: Modified, thanks for your suggestion!

**Comment by gbaned** [comment]
@Zantares Could you please resolve the conflicts? Thanks!

**Review by penpornk — DISMISSED** [review]
Ouch. For some reason, I only saw that you have updated this just now. So sorry for the delay! (I'll re-approve if the merging has conflicts.)

**Comment by penpornk** [comment]
@Zantares Could you please help resolve the conflicts? Thank you!

**Comment by Zantares** [comment]
> @Zantares Could you please help resolve the conflicts? Thank you!

Sorry for the delay, I was on a vacation few days ago. I have merged the master to this branch and fixed the conflict, please take a review again, thanks!

**Comment by Zantares** [comment]
This PR helps me to find redundant attributes in UT,  I removed them and made a new commit.

**Review by penpornk — APPROVED** [review]
No worries! Thanks again for your PR!

**Pull request merged**